### ISSUE TITLE
THiGER loader: KGLinkNeighborLoader yields batch-local indices during iteration (labels attach to the wrong entities)

### ISSUE THREAD

**Issue by hfffghy** [SonyResearch/HakkenOSS#2]
**Component:** `models/packages/hakken-models/hakken_models/data_loaders/kg_link_neighbor_loader.py`
(`KGLinkNeighborLoader`), used by `THiGERDataModule` for training and validation.

**Summary.** The loader converts `edge_label_index` from batch-local to global ids in an overridden
`__call__`. PyG's `LinkLoader` never calls `__call__` while iterating: its iterator runs
`collate_fn` and then `filter_fn` (`_get_iterator` → `DataLoaderIterator(..., self.filter_fn)`). This
holds in torch-geometric 2.7.0 (the version in `hakken-models/uv.lock`) and 2.8. Every iterated
batch therefore carries LOCAL indices, and `THiGER.forward` reads them as global entity ids: labels
attach to the wrong entities.

**Reproduction.** A 60-node graph and 6 target pairs with one-hot labels; iterate the loader and
compare each yielded `(edge_label_index, edge_label)` with the input pairs: 0 of 6 are aligned.
Script (run from any directory; pass the path that contains `hakken_models`):

```python
"""R-H5 check, run inside `hakken_pyg` against a given hakken_models tree.

KGLinkNeighborLoader (the THiGER loader) must yield target pairs in GLOBAL entity ids when ITERATED,
as the model reads them. The released code maps ids in __call__, which PyG's iterator never calls.
Prints JSON: how many iterated target pairs are the true global pairs, and how many are not.
Usage: python hakken_rh5_check.py <path containing hakken_models>
"""
import json
import sys
import typing

import typing_extensions

sys.path.insert(0, sys.argv[1])
for _n in ("Self", "StrEnum"):
    if not hasattr(typing, _n) and hasattr(typing_extensions, _n):
        setattr(typing, _n, getattr(typing_extensions, _n))

import torch                                                                   # noqa: E402
from hakken_models.core.entities.kg_data import KGData                         # noqa: E402
from hakken_models.data_loaders.kg_link_neighbor_loader import KGLinkNeighborLoader  # noqa: E402

g = torch.Generator().manual_seed(0)
n = 60
src = torch.randint(0, n, (400,), generator=g); dst = torch.randint(0, n, (400,), generator=g)
data = KGData(x=torch.zeros(n, 1), edge_index=torch.stack([src, dst]),
              edge_attr=torch.stack([torch.zeros(400, dtype=torch.long), torch.randint(0, 5, (400,), generator=g)], 1),
              num_nodes=n)
pairs = torch.tensor([[40, 7], [13, 55], [2, 31], [59, 0], [21, 44], [8, 36]])
labels = torch.eye(6)[:, :4]
truth = {tuple(p): tuple(l.tolist()) for p, l in zip(pairs.tolist(), labels)}
loader = KGLinkNeighborLoader(data=data, num_neighbors=[5, 5], edge_label_index=pairs.t().contiguous(),
                              edge_label=labels, batch_size=3, shuffle=False)
aligned = misaligned = 0
for b in loader:
    for p, l in zip(b.edge_label_index.t().tolist(), b.edge_label.tolist()):
        if truth.get(tuple(p)) == tuple(l):
            aligned += 1
        else:
            misaligned += 1
print(json.dumps({"aligned": aligned, "misaligned": misaligned}))
```

Output on the current `main`: `{"aligned": 0, "misaligned": 6}`; with the fix below: `{"aligned": 6, "misaligned": 0}`.

**Impact.** A 512-pair overfit test stays at the marginal loss under any learning rate, batch size,
Pre-LN transformer or warmup; with the fix it reaches weighted F1 0.85 in 10 epochs. On a
PubTator/CTD slice, the released model predicts label base rates (macro F1 .078–.117 against base
rates .077–.085 at cutoffs 1990/2000/2010), and the fixed one reaches .23–.32. Published THiGER
numbers produced with this data module should be re-read with this in mind.

**Fix.** Move the remapping into `filter_fn`, which every path (iterator and `__call__`) runs once:

```python
def filter_fn(self, out):
    raw_batch = super().filter_fn(out)
    raw_batch.edge_label_index = raw_batch.n_id[raw_batch.edge_label_index]
    return raw_batch
```
(and delete the `__call__` override, which would otherwise map twice).

_Found while evaluating THiGER on a PubTator/CTD slice; environment: torch-geometric 2.8.0 (behaviour identical in 2.7.0's `LinkLoader._get_iterator`), torch 2.2.1, Python 3.10._
